libs/model/src/aggregates/user/SignIn.command.ts

```diff
@@ -162,17 +169,26 @@ export function SignIn(): Command<typeof schemas.SignIn> {
                 created_at: addr.created_at!,
               },
             });
-          (new_address || !wallet_found) &&
+          if (new_address || !wallet_found) {
+            const balances = eth_chain_id
```

Comment:
Adding the required balance fetching to sign in doesn't seem right. It tightly couples our sign-in flow with Alchemy or whatever chain node is associated with the community. This is especially troublesome for eth chains that have poor RPC nodes. It seems that this is used to award XP so maybe this should be optional i.e. it's better to award lower XP than to fail sign in completely. Other option could be a separate job to award XP after the fact (with retries).